### PR TITLE
Add support for Metal arrays with `MtlCellArray`

### PR DESCRIPTION
**Pull request by GiackAloZ**
This PR adds support for Metal arrays in CellArray.jl, introducing the type `MtlCellArray` similarly to other GPUArrays.jl compatible backends.

### CODE REVIEW

File: test/test_CellArray.jl [hunk 1]
```diff
@@ -53,30 +62,34 @@ mutable struct MyMutableFieldArray{T} <: FieldArray{Tuple{2}, T, 1}
     yxxx::T
 end
 
+const TEST_PRECISIONS = [Float32, Float64]
+for (array_type, Array, CellArray, allowscalar) in zip(array_types, ArrayConstructors, CellArrayConstructors, allowscalar_functions)
```
Comment (omlins) [inline test/test_CellArray.jl:66]: Doesn't it change the output when you move the loop outside of everything?

File: test/test_CellArray.jl [hunk 2]
```diff
@@ -53,30 +62,34 @@ mutable struct MyMutableFieldArray{T} <: FieldArray{Tuple{2}, T, 1}
     yxxx::T
 end
 
+const TEST_PRECISIONS = [Float32, Float64]
```
Comment (omlins) [inline test/test_CellArray.jl:65]: @albert-de-montserrat could you integrate the changes you did in your PR that didnt fully work unfortunately in the tests concerning the float type?

**Review by omlins — CHANGES_REQUESTED** [review]
@albert-de-montserrat could you integrate the changes you did in your PR that didnt fully work unfortunately in the tests concerning the float type?

File: test/test_CellArray.jl (hunk 1 above)
Comment (GiackAloZ) [reply test/test_CellArray.jl:66]: Yeah so now all the tests with different array type and precision are launched on different testsets.

Comment (GiackAloZ) [reply test/test_CellArray.jl]: Loop restored as it was

**Comment by GiackAloZ** [comment]
@omlins requested changes in tests have been implemented, requesting another review.

**Review by omlins — APPROVED** [review]
LGTM

**Comment by omlins** [comment]
Thanks @GiackAloZ  and @albert-de-montserrat !

**Comment by omlins** [comment]
The CI fails for Windows and MacOS:
https://github.com/JuliaGPU/Metal.jl/issues/457
https://github.com/JuliaGPU/Metal.jl/issues/458

Merged nevertheless into main.

**Pull request merged**